ciao/arch/architecture.md

```diff
@@ -423,3 +423,88 @@ various network constructs such as subnets and interfaces are assigned names
 from different namespaces by both the Cloud Integrated Advanced Orchestrator
 and docker.  The ciao network driver maintains a database so that it can
 translate identifiers from one namespace to another.
+
+## Storage
+
+The Cloud Integrated Advanced Orchestrater supports two forms of storage for instances:
+
+1. local storage, in which an instance's data, including its rootfs, is stored
+on the disk of the node upon which the instance is hosted
+2. ceph based storage, in which an instance's data is stored in the
+ceph cluster.
+
+{% include important.html content="Local storage is only available for
+containers." %} Both the rootfs and any disk based data written or read by a VM
+instance must be stored in a ceph volume.  This is in contrast to containers,
+whose root file systems are always stored on the node on which the container is hosted.
+Processes running within these containers can write to the containers' root file
+systems.  Although, disk based IO will be very fast in containers that
+read and write to their rootfs, there is a downside.  Any data written to a
+local volume does not survive migration.  The rootfs of a container is reset to
+its default state when it is migrated from one node to another.  It is however
+possible to create a data volume inside the ceph cluster and attach this volume
+to a container.  Any data written to this volume by the container will survive
+migration.
+
+### Images
+
+The controller implements an image service.  This service manages a set of of
+RBD (Rados Block Device) volumes that serve as backing images for the root file
+systems of newly created VM instances.  The image service allows users to
+create, delete, inspect and enumerate images.  Normal users can create tenant
+private images, i.e., images that are associated with a given tenant and can
+only be used to create root file systems for instances belonging to the same
+tenant.  Administrators can create public images, that that can be used by all
+users to create VM instances in any tenants they like.
+
+When a new image is uploaded to the image service it is converted into a raw
+format and stored in a new RBD volume in the ceph cluster.  An RBD snapshot is
+then taken of this new volume and the snapshot is protected.  It is this
+protected RBD snapshot that is used as the backing image for VM instances.  The
+image will be assigned a UUID when created.  It is also possible for users to
+specify a friendly name at creation time.
+
+In order to create an instance that uses an image in the image service as the
+backing image for its rootfs, the backing image must be referenced in the workload
+definition used to create the instance.  The backing image can be referenced by
+UUID or by its friendly name.  When the controller creates the new VM, it first
+creates its rootfs, by cloning the protected snapshot that represents the backing
```

Comment:
can you make it clear that the clone is shallow with copy-on-write semantics for the blocks that make it up.